efel/pyfeatures/pyfeatures.py

```diff
@@ -28,39 +28,62 @@
 
 import numpy
 import efel.cppcore
+from scipy.ndimage.filters import gaussian_filter1d
+from numpy.fft import *
 
 
 all_pyfeatures = [
     'voltage',
     'time',
-    'current',
     'ISIs',
     'initburst_sahp',
     'initburst_sahp_vb',
     'initburst_sahp_ssse',
-    'depol_block']
+    'depol_block',
+    'current',
+    'impedance']
 
 
 def voltage():
     """Get voltage trace"""
     return _get_cpp_feature("voltage")
 
 
+def current():
+    """Get voltage trace"""
+    return _get_cpp_feature("current")
+
+
 def time():
     """Get time trace"""
     return _get_cpp_feature("time")
 
 
-def current():
-    """Get current trace"""
-    return _get_cpp_feature("current")
+def impedance():
+    volta = _get_cpp_feature("voltage")
+    volt_hold = numpy.median(volta[0:10])
+    volt = volta - volt_hold
+    curr = _get_cpp_feature("current")
```

Comment:
check also if returned value is not None (mostly current is not defined for a trace)

Reply:
What do we do if the returned value for "current" is None?

Reply:
Send a warning message and brake

Reply:
eFEL should never crash, because it could stop entire optimization runs. So an option is to return None. The C++ code has a variable 'GErrorStr' that contains warnings (which can be silenced). I don't think this variable can be accessed from here, but maybe you can check.